### PR TITLE
LPD-22652 Use locks to avoid updating the same object action concurrently

### PR DESCRIPTION
**Pull request by devpedrotavares**
Related ticket: [LPD-22652](https://liferay.atlassian.net/browse/LPD-22652)

### CODE REVIEW

**Review by marcelabc — CHANGES_REQUESTED** [review]
Great work in developing this solution; it was a challenge! :clap: 
Just a minor change in my review.

**Comment by devpedrotavares** [comment]
ci:test:sf

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectActionLocalServiceImpl.java [hunk 1]
```diff
@@ -391,15 +394,37 @@ public ObjectAction updateObjectAction(
...
+			ObjectAction objectAction =
+				objectActionPersistence.findByPrimaryKey(objectActionId);
+
+			objectAction.setStatus(status);
+
+			return objectActionPersistence.update(objectAction);
+		}
+		finally {
+			LockManagerUtil.unlock(
+				ObjectAction.class.getName(), String.valueOf(objectActionId));
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectActionLocalServiceImpl.java:426]: I believe this method has an option for the second parameter to be `long`. Can you check that, please?

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

**Comment by devpedrotavares** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:forward:force

**Pull request closed without merging**